README.md

```diff
@@ -431,6 +431,97 @@ Certain WebVR `Gamepad` objects have some components with an inverted `yAxis` ca
+
+### WebXR Connection and Disconnection
+This sample code shows creating an `XRGamepad` based on the `XRInputSource.gamepad` property for all newly connected `XRInputSource` objects.  When an `XRInputSource` is disconnected the `XRGamepad` is released.
+
+```js
+let xrGamepads = {};
+xrSession.addEventListener('inputsourceschange', onInputSourcesChange);
+
+function onInputSourcesChange(event) {
+  event.added.forEach((inputSource) => {
+    if (inputSource.gamepad) {
+      let mapping = XRGamepad.getMapping(gamepad.id, XRMappingConstants.MappingType.WebXR);
+      let xrGamepad = new XRGamepad(gamepad, mapping, inputSource.handedness);
+      xrGamepads[inputSource] = xrGamepad;
+    }
+  }
+
+  event.removed.forEach((inputSource) => {
+    if (xrGamepads[inputSource]) {
+      delete xrGamepads[inputSource];
+    }
+  });
+}
+```
+
+### WebVR Connection and Disconnection
+This sample code shows creating an `XRGamepad` for all newly connected `Gamepad` objects associated with WebVR.  When an associated `Gamepad` is disconnected the `XRGamepad` is released.
+
+```js
+let xrGamepads = {};
+navigator.addEventListener('gamepadconnected', onGamepadConnected);
+navigator.addEventListener('gamepaddisconnected', onGamepadDisconnected);
+
+function onGamepadConnected(event) {
+  let gamepad = event.gamepad;
+  if (gamepad.deviceId) {
+    let mapping = XRGamepad.getMapping(gamepad.id, XRMappingConstants.MappingType.WebVR);
+    let xrGamepad = new XRGamepad(gamepad, mapping, gamepad.handedness);
+    xrGamepads[gamepad] = xrGamepad;
+  }
+}
+
+function onGamepadDisconnected(event) {
+  if (xrGamepads[event.gamepad]) {
+      delete xrGamepads[event.gamepad];
+    }
+  }
+}
+```
+
+## Engine Loop
+On each frame, the gamepad data must be requeried and the engine must respond accordingly to the new data.  This will be the same for both WebXR and WebVR
+
+```js
+function updateGamepads() {
+  Object.key(xrGamepads).forEach((key) => {
+    let xrGamepad = xrGamepads[key];
+    xrGamepads.components.forEach((xrComponent) => {
+      const xrComponentData = xrComponent.getData();
+      MyEngine.respondToInput(key, xrComponentData);
+
+      const weightedVisualizations = xrComponent.getWeightedVisualizations();
+      MyEngine.updateInputVisuals(key, weightedVisualizations);
+    });
+  });
+}
+```
+
+> TODO Add an explanation for how to updated the visuals based on the weighted visualizations
+
+## Target Ray Origin
```

Comment:
So this is feedback that's a couple of PRs too late, but it didn't really click with me till I saw it used in context. The use of "origin" here feels to me like it only implies a position offset, so it read kind of weird to me at first. Is there a better term that we could use?

Reply:
I'm open to whatever, though I'd like to not hold this PR for it...